api/src/services/hashnode.ts

```diff
@@ -1,13 +1,43 @@
 import { request } from 'graphql-request'
-import gql from 'graphql-tag'
+import type { Post, Publication, QueryrecentPostsArgs } from 'types/graphql'
 
-export const recentPosts = async () => {
-  const query = gql`
+import { logger } from 'src/lib/logger'
+
+type HashnodePost = Post
+
+type Edge<T> = {
```

Comment:
Likely move into some shared set of types and this generic can be used for other edges.